model_analyzer/record/gpu_memory_record.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+from model_analyzer.record.gpu_record import GPURecord
+
+
+class GPUMemoryRecord(GPURecord):
+    """
+    GPUMemoryRecord stores information related to
+    GPU memory usage
+    """
+
+    def __init__(self, device, used, total, free):
+        super().__init__('memory', device)
```

Comment:
Is there an advantage to having this `name` property in the super class that can't be acheived with checking `isinstance` at the point where `name` is required? In `test_record_memory` you seem to use the latter.

Reply:
@xprotobeast2 I see, you mean I remove the `name` property and use the `isnstance` to determine the type of the memory record?

Reply:
Yes, was just wondering in what situation you'd want a string instead.

Reply:
Assume that we want to create a output writer later on the parse the results in each record. I think it would a bit awkward to use `isinstance` to find the record type.